index.js

```diff
@@ -18,48 +18,70 @@ StylusFilter.prototype.constructor = StylusFilter;
 StylusFilter.prototype.extensions = ['styl'];
 StylusFilter.prototype.targetExtension = 'css';
 
-StylusFilter.prototype.processString = function (str) {
+StylusFilter.prototype.processString = function(str) {
 	var opts = this.options;
+
+	var importRegex = new RegExp(/@import ['"].*['"]/);
+	while (importRegex.test(str)) {
+		var imp = importRegex.exec(str);
+		var sourceReg = new RegExp(/['"].*['"]/);
+		var extReg = new RegExp(/.styl/);
+		var src = sourceReg.exec(imp[0])[0].replace(/"/g, '').replace(/'/g, '');
+		if (!extReg.test(src)) {
+			src = src + '.styl';
+		}
+
+		src = this.inputTree.inputTree.inputTree + '/' + this.inputTree.srcDir + '/' + src;
```

Comment:
I'm not asking you to remove it. I'm asking what it does and why. Some comments on would be useful.